engine/src/main/java/io/camunda/zeebe/engine/state/immutable/DeploymentState.java

```diff
@@ -30,7 +30,7 @@ public interface DeploymentState {
    */
   boolean hasPendingDeploymentDistribution(long deploymentKey, int partitionId);
 
-  DeploymentRecord getStoredDeploymentRecord(long deploymentKey);
+  DeploymentRecord getStoredDeploymentRecord(long deploymentKey, String tenantId);
```

Comment:
:thought_balloon: For the sake of transparency, in the `DeploymentState` API, we're not providing any methods that would accept a List of (the user's) authorized tenants since there won't be anything in the `DeploymentState` to check for, i.e. we will perform the authorization check before the state is touched at all.